project/tests/test_utils.gd

```diff
@@ -0,0 +1,136 @@
+
+var time_elapsed = 0
+var keys_typed = []
+var quit = false
+var expected_ref_props = ['Reference', 'script', 'Script Variables']
+var custom_ref_props = ['profile_name', 'environment', 'slots']
+var dirA = "res://tests/files/dirA"
+var dirB = "res://tests/files/dirB"
+var dirD = "res://tests/files/dirD"
+var json1 = "res://tests/files/dirA/json1"
+var json2 = "res://tests/files/dirA/json2"
+var json3 = "res://tests/files/dirA/json3"
+
+func _initialize():
+	print("Initialized:")
+	print("  Starting time: %s" % str(time_elapsed))
+
+
+func _finalize():
+	print("Finalized:")
+	print("End time: %s" % str(time_elapsed))
+	print("Keys typed: %s" % var2str(keys_typed))
+
+func test_copy_dir():
+#validate that the directory copy is successful
+	var is_copied = Util.copy_dir(dirA, dirB)
+	assert(is_copied == true)
+
+
+func test_read_json_file():
+#validate that the json read is successful
+	var json_file = Util.read_json_file(json1)
+	print(json_file)
+	var p = JSON.parse(json_file)
+	assert(p.result == TYPE_ARRAY)
+
+
+# test name starts with same name e.g. test_b vs test_b2
+func test_err():
+	var error = Util.err("This is an error. Duh!")
+	print(error)
+	assert(error is Reference)
```

Comment:
should assert if the GDResult is actually an error (and not ok)

Reply:
yepp, I got that I was not sure how to do that at first